routes/offering_test.go

```diff
@@ -0,0 +1,402 @@
+//	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("/offerings?%s&%s", offeringStart, offeringEnd), nil)
+//	if err != nil {
+//		t.Fatal(err)
+//	}
+//	q := req.URL.Query()
+//	q.Add("start", offeringStart)
+//	q.Add("end", offeringEnd)
+//	req.URL.RawQuery = q.Encode()
+//	//req = mux.SetURLVars(req, config.URLParams)
+//	rr := httptest.NewRecorder()
+//	handler := http.HandlerFunc(suite.app.GetOfferingsByDateRange)
+//	handler.ServeHTTP(rr, req)
+//	assert.Equal(t, rr.Code, http.StatusOK, "status code")
+//	var payload []*model.Offering
+//	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+//	assert.Equal(t, 1, len(payload), "testGetOfferingsByDateRange: incorrect response size")
+//	expectedOfferings := []*model.Offering{Offering5}
+//	for _, expected := range expectedOfferings {
+//		found := false
+//		for _, o := range payload {
+//			if o.Equal(expected) {
+//				found = true
+//			}
+//		}
+//		if !found {
+//			t.Fatalf("testGetOfferingsByDateRange: %s not found in offring list", expected.ID)
+//		}
+//	}
+//}
+
+var startNew, _ = utils.TimeStampToTime("1583884800")
+var endNew, _ = utils.TimeStampToTime("1583971199")
+var newOffering = &model.Offering{
+	ID: "", // Unknown at this point
+	UserID: "32ea2fb1-7124-304a-b9c3-eb445578103e",
+	WorkspaceID: "5e56de3d-2323-372d-897f-23d6037c8581",
+	Cancelled: false,
+	StartDate: startNew,
+	EndDate: endNew,
+	CreatedBy: "8b5bb736-6a1d-3378-8e71-ab45fe8beb84",
+}
+
+func (suite *AppTestSuite) TestCreateOffering() {
+	// Check ID does not exist in database
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetAllOfferings,
+		URL:     "/offerings/",
+	})
+	var payload []*model.Offering
+	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+	for _, o := range payload {
+		if o.EqualMinusID(newOffering) {
+			t.Fatalf("testCreateOffering: %s found in offering list, when it should not exist", newOffering.ID)
+		}
+	}
+	// [POST] Create offering
+	requestBody, _ := json.Marshal(map[string]interface{}{
+		"workspace_id": newOffering.WorkspaceID,
+		"user_id": newOffering.UserID,
+		"start_time": newOffering.StartDate,
+		"end_time": newOffering.EndDate,
+		"cancelled": newOffering.Cancelled,
+		"created_by": newOffering.CreatedBy,
+	})
+	rr2 := executeReq(t, &testRouteConfig{
+		Method:  http.MethodPost,
+		Body: bytes.NewBuffer(requestBody),
+		Handler: suite.app.CreateOffering,
+		URL:     fmt.Sprintf("/offerings"),
+	})
+	// Check correct response
+
+	assert.Equal(t, rr2.Code, http.StatusCreated, "status code")
+	var payload2 *model.Offering
+	_ = json.Unmarshal(rr2.Body.Bytes(), &payload2)
+	// Just give it the ID since we want to use assert
+	newOffering.ID = payload2.ID
+	assert.Equal(t, newOffering, payload2, "The created offering is not the same as the sent request.")
+}
+var startPatch, _ = utils.TimeStampToTime("7952342400")
+var endPatch, _ = utils.TimeStampToTime("7952515199")
+var patchOffering = &model.Offering{
```

Comment:
if its not being used outside the `TestPatchOffering` function i would just make this a local variable